nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -80,6 +80,8 @@
 
 - The Caddy module gained a new option named `services.caddy.enableReload` which is enabled by default. It allows reloading the service instead of restarting it, if only a config file has changed. This option must be disabled if you have turned off the [Caddy admin API](https://caddyserver.com/docs/caddyfile/options#admin). If you keep this option enabled, you should consider setting [`grace_period`](https://caddyserver.com/docs/caddyfile/options#grace-period) to a non-infinite value to prevent Caddy from delaying the reload indefinitely.
 
+- The `krb5` module has been rewritten, removing `krb5.libdefaults`, `krb5.realms`, `krb5.domain_realm`, `krb5.capaths`, `krb5.appdefaults` and `krb5.plugins`, which have been replaced by `krb5.settings`. Additionally, `krb5.config` has been replaced by `krb5.extraConfig`.
```

Comment:
extraConfig is the old mulitline string setting name. Can't we get rid of that, too?

Reply:
There are some parts of this file which don't follow the normal format such as `module` or `include`/`includedir` statements (sure, these could be added as options instead but I assume they're very rarely used since they aren't in the current module either so I left them).

Plus this way config generation is kept extensible in case other modules (user configurations) want to write custom text into the file. I don't see the harm in exposing it since it doesn't add any extra maintenance overhead, everything needs to be formatted as text either way, and I've hit situations before where something like this wasn't exposed but instead a variable in the module and I had to use a custom nixpkgs to add what I wanted.